### PR TITLE
[DEV-11849] & [DEV-11850] Add Hero and StoryCard component to storybook

### PR DESCRIPTION
**Pull request by fgyimah**

### CODE REVIEW

File: packages/ui/src/components/StoryCard/StoryCard.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                size === 'large' ? `flex-col gap-0` : `md:flex-row items-start gap-4`,
+                className,
+            )}
+        >
+            <Link href={`/${slug}`} locale={false} legacyBehavior>
+                <div
+                    className={twMerge(
+                        size === 'large'
+                            ? 'w-full aspect-[20/14] md:aspect-[27/17]'
+                            : 'w-1/3 aspect-[10/7] hidden md:block',
```
Comment (e1himself) [inline packages/ui/src/components/StoryCard/StoryCard.tsx:48]: Isn't `10/7` the same as `20/14`?

File: packages/ui/src/components/StoryPublicationDate/StoryPublicationDate.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export function StoryPublicationDate({ story, locale }: Props) {
+    const date = getStoryPublicationDate(story);
+
+    if (!date) {
+        return null;
+    }
+
+    return (
+        <>{date.toLocaleDateString(locale, { year: 'numeric', month: 'long', day: 'numeric' })}</>
+    );
```
Comment (e1himself) [inline packages/ui/src/components/StoryPublicationDate/StoryPublicationDate.tsx:18]: I believe date formats should be customizable. 

We have this setting available in the app:

image

File: packages/ui/src/components/CategoryLink/CategoryLink.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import type { Category, Culture } from '@prezly/sdk';
+import { getCategoryUrl, getLocalizedCategoryData, LocaleObject } from '@prezly/theme-kit-core';
+import Link from 'next/link';
+import { twMerge } from 'tailwind-merge';
+
+interface Props {
+    category: Category;
```
Comment (e1himself) [inline packages/ui/src/components/CategoryLink/CategoryLink.tsx:7]: It is recommended to `Pick<...>` only the properties the component needs to make it possible using the library with a wider range of `@prezly/sdk` package versions.

Read more about it here: https://github.com/prezly/theme-kit-js#prezly-sdk-dependency

File: packages/ui/src/components/StoryCard/StoryCard.tsx (hunk 1 above)
Comment (fgyimah) [reply packages/ui/src/components/StoryCard/StoryCard.tsx:48]: Is the same, but the large has a different aspect ration from tablet to desktop viewport

Comment (e1himself) [reply packages/ui/src/components/StoryCard/StoryCard.tsx:48]: I'm just wondering why the same aspect ratio is used with two different notations. `20/14` can be replaced with `10/7` with no side effects. Right?

Comment (fgyimah) [reply packages/ui/src/components/StoryCard/StoryCard.tsx:48]: Yeah right

Comment (fgyimah) [reply packages/ui/src/components/StoryCard/StoryCard.tsx]: Updated https://github.com/prezly/theme-kit-js/pull/414/commits/7f5e4071f2ebd39b7cae32883fec555728b1ac8b

File: packages/ui/src/components/StoryPublicationDate/StoryPublicationDate.tsx (hunk 2 above)
Comment (fgyimah) [reply packages/ui/src/components/StoryPublicationDate/StoryPublicationDate.tsx]: Updated https://github.com/prezly/theme-kit-js/pull/414/commits/7f5e4071f2ebd39b7cae32883fec555728b1ac8b

File: packages/ui/src/components/CategoryLink/CategoryLink.tsx (hunk 3 above)
Comment (fgyimah) [reply packages/ui/src/components/CategoryLink/CategoryLink.tsx]: Updated https://github.com/prezly/theme-kit-js/pull/414/commits/7f5e4071f2ebd39b7cae32883fec555728b1ac8b

File: packages/ui/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import type { Story } from '@prezly/sdk';
+
+export type StoryWithImage = Story & Pick<Story.ExtraFields, 'thumbnail_image'>;
```
Comment (oleg-semyonov) [inline packages/ui/src/types.ts:3]: Hey @e1himself, I wonder if we should include the `thumbnail_image` into base `Story` object at some point. We're using this field in every theme, and this type we've been using feels a bit like a hack 🤔

**Review by oleg-semyonov — COMMENTED** [review]
Looks good! 👍🏻

File: packages/ui/src/types.ts (hunk 4 above)
Comment (e1himself) [reply packages/ui/src/types.ts:3]: >  I wonder if we should include the thumbnail_image into base Story object at some point. 

Hey Oleg! 

I'm not sure yet. The thumb rule for picking defaults for Stories API response fields is whether it is needed on the listing page (in the app), and `thumbnail_image` is not needed there.

But I see your point. I'll need to think about it.

**Pull request merged**